src/Guide/Api/Methods.hs

```diff
@@ -65,10 +67,16 @@ setCategoryNotes :: DB -> Uid Category -> Text -> Handler NoContent
 setCategoryNotes db catId note =
   dbQuery db (GetCategoryMaybe catId) >>= \case
     Nothing -> throwError (err404 {errBody = "Category not found"})
-    Just _ -> do
-      (_edit, _newCategory) <- dbUpdate db (SetCategoryNotes catId note)
-      -- TODO diff and merge
-      pure NoContent
+    Just original -> do
+      let fromMarkDown = markdownBlockMdSource . _categoryNotes
+      modified <- dbQuery db (GetCategory catId)
+      if fromMarkDown original /= fromMarkDown modified then do
+        let merged = merge (fromMarkDown original) note (fromMarkDown modified)
+        (_edit, _newCategory) <- dbUpdate db (SetCategoryNotes catId merged)
```

Comment:
Nope. We don't simply put merged content into the database, because the merge might be crappy.

The merged version (along with the `original`, `modified` and `note`) should be returned in the response, with status code 409 conflict. You can create a client type `CMergeConflict` for that.

Reply:
Don't forget to update the other handler, too.